traffic_portal/app/src/common/modules/table/cacheGroupParameters/TableCacheGroupParamsUnassignedController.js

```diff
@@ -17,13 +17,15 @@
  * under the License.
  */
 
-var TableCacheGroupParamsUnassignedController = function(cg, parameters, $scope, $uibModalInstance) {
+var TableCacheGroupParamsUnassignedController = function(cg, allParams, assignedParams, $scope, $uibModalInstance) {
 
 	var selectedParams = [];
 
 	$scope.cg = cg;
 
-	$scope.unassignedParams = parameters;
+	$scope.unassignedParams = _.filter(allParams, function(param) {
+		return !_.find(assignedParams, function(assignedParam) { return assignedParam.id == param.id });
```

Comment:
Instead of using `_.find`, consider using the Javascript built-in [`Array.find`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) method.

Also, though, you could avoid iterating the whole array of assigned params if it's a Set of ids first:
```javascript
// In TableCacheGroupParametersController.js
// ...
assignedParams: function() {
	return new Set(parameters.map(x=>x.id));
	// I think maybe webpack won't actually handle fat-arrow functions, so might need to do:
	// return new Set(parameters.map(function(x){return x.id;}));
}
//...
// now in TableCacheGroupParamsUnassignedController.js
$scope.unassignedParams = allParams.filter(x=>!assignedParams.has(x.id));
// or, if you can't use "fat-arrow" functions:
$scope.unassignedParams = allParams.filter(
	function(p) {
		return !assignedParams.has(p.id);
	}
);
```

Reply:
sure ill look into it